python-package/xgboost/core.py

```diff
@@ -706,10 +629,8 @@ def set_group(self, group):
         group : array like
             Group size of each group
         """
-        if _has_cuda_array_interface(group):
-            self.set_interface_info('group', group)
-        else:
-            self.set_uint_info('group', group)
+        from .data import dispatch_meta_backend
```

Comment:
Any reason not to use a top-level import?

Reply:
Will change it to top level.

Reply:
Actually no.  I did this because it can avoid circular module import.